### ISSUE TITLE
Restore assumes "1.0.0" version for P2P that calculates its version by build

### ISSUE THREAD

**Issue by AArnott** [AArnott/Nerdbank.GitVersioning#113]
**Repro steps**

1. Create two .netstandard libraries: A and B with a common solution directory.
1. Add the solution to git source control and commit.
1. Add reference from B to A
1. Install [Nerdbank.GitVersioning package](https://nuget.org/packages/nerdbank.gitversioning) to both
1. Add a version.json file to the solution folder with content: `{ "version": "0.1-beta" }`
1. Commit all changes to git.
1. Run a package restore on the solution and pack both projects.

**Expected**

The **B** nuget package references **A** package by its version (e.g. `0.1.2-beta`)

**Actual**

The **B** nuget package references **A** package by a non-existent version `1.0.0`

d15rtwsvc 26228.4

Blocked by NuGet/Home#4790

**Comment by AArnott** [comment]
Hacky workaround: add this to the referenced project:

```xml
  <Target Name="FixUpVersion" BeforeTargets="_GenerateRestoreProjectSpec"
      DependsOnTargets="GetBuildVersion" />
  <ImportGroup Condition=" '$(ExcludeRestorePackageImports)' == 'true' ">
    <Import Project="$(UserProfile)\.nuget\packages\nerdbank.gitversioning\1.6.19-beta\buildCrossTargeting\Nerdbank.GitVersioning.targets"
            Condition="Exists('$(UserProfile)\.nuget\packages\nerdbank.gitversioning\1.6.19-beta\buildCrossTargeting\Nerdbank.GitVersioning.targets')" />
  </ImportGroup>
```

**Comment by onovotny** [comment]
That's nasty, bit I'll give it a shot

**Comment by onovotny** [comment]
Just applied the hack here: https://github.com/Reactive-Extensions/Rx.NET/blob/14287b166479f770408b6f760b71b942a7fb99ef/Rx.NET/Source/Directory.build.props#L38-L50

Soooo gross.

**Comment by AArnott** [comment]
Note, MSBuild 15.6 broke the workaround. It interestingly enough works again if you update to the latest version of NB.GV (2.1.23). I'm not sure what changed between 1.6.25 and 2.1.23 to make the workaround effective on 15.6 agents, but everyone should update!

**Comment by bording** [comment]
@AArnott I believe 15.6 includes the new functionality added in https://github.com/NuGet/NuGet.Client/pull/1915 and https://github.com/AArnott/Nerdbank.GitVersioning/pull/161, so you shouldn't need the workaround any more.

**Comment by AArnott** [comment]
Oh ya, good point. That is probably why the upgrade fixes it. By leaving the workaround in place in the referencing project, it should work properly in 15.5 and 15.6 (Appveyor still has 15.5 from what I can tell). 

It's too bad that in the fix in 15.6, they broke the workaround.

**Comment by AArnott** [comment]
No workaround is required with VS 2017 Update 6 (Dev15.6) as they've fixed the issue and NB.GV has adapted to leverage the new feature.

Just make sure you're on NB.GV 2.1.23 or later.
